src/ucp/core/ucp_context.c

```diff
@@ -1939,6 +1957,14 @@ ucs_status_t ucp_init_version(unsigned api_major_version, unsigned api_minor_ver
         context->rcache = NULL;
     }
 
+    /* Update registration memory domain map for host memory type taking into
+     * account result of ucp_memh_alloc. */
+    status = ucp_context_reg_md_map_alloc(context);
```

Comment:
maybe move this inside ucp_fill_resources, after the call to ucp_add_component_resources, which initializes reg_md_map.